complete recursion maintenance (#3692)
cpinitiative/usaco-guide: content/2_Bronze/Complete_Rec.mdx

using ll = long long;

int n;
vector<long long> weights;

ll recurse_apples(int index, ll sum1, ll sum2) {

> 	// We've added all apples- return the absolute difference

Reviewer: ```suggestion
	// We've added all apples - return the absolute difference
```

Reviewer: oh wait ryan requested the same thing bruh i didnt see

weights = list(map(int, input().split()))

def recurse_apples(i: int, sum1: int, sum2: int) -> int:

> 	# We've added all apples- return the absolute difference

Reviewer 2: ```suggestion
	# We've added all apples - return the absolute difference
```

Author: does this really matter? if you insist i'll change it

</Warning>

A **bitmask** is an integer whose binary representation is used to represent a

> subset. For a particular bitmask, if the $i$'th bit is equal to $1$,

Reviewer: ```suggestion
subset. In the context of this problem, if the $i$'th bit is equal to $1$ in a particular bitmask,
```

</PySection>
</LanguageSection>

> Note the fancy bitwise operations:

Reviewer: why did you move this down here?

Author: intended flow imo should be
"oh alr look @ the code"
"wtf are these things"
scrolls down
"oh alr"

Reviewer: nah what if they dont even realize its there
its awkward to put it all the way down there too
whats the advantage of this? They might figure out what the operation is doing by themselves? They're pre-bronze contestants, after all.

Author: fair

```cpp
#include <bits/stdc++.h>
using namespace std;

string s;
vector<string> perms;
int char_count[26];

> void search(const string &curr = "") {

Reviewer: original was more intuitive bruh

Author: my man i changed basically nothing

Reviewer: then dont change it

Author: ooo references, how scary

Reviewer: why add when you could just not

Author: it's good practice to pass references when you can
my very first pr on this site taught me that
